### PR TITLE
fix: accept legacy flight requests

### PR DESCRIPTION
**Pull request by crepererum**
This removes the SERVER-side breaking change introduced in #4273 by also
accepting JSON-based tickets.

Closes #4280.

### CODE REVIEW

File: service_grpc_flight/src/lib.rs [hunk 1]
```diff
@@ -111,6 +125,34 @@ impl Error {
...
 
+#[derive(Deserialize, Debug)]
+/// Body of the `Ticket` serialized and sent to the do_get endpoint.
+struct ReadInfo {
+    database_name: String,
+    sql_query: String,
+}
+
+impl ReadInfo {
+    fn decode_json(ticket: &[u8]) -> Result<Self> {
```
Comment (alamb) [inline service_grpc_flight/src/lib.rs:136]:
```suggestion
    /// decode "legacy" format ReadInfo that was encoded as JSON (not protobuf)
    fn decode_json(ticket: &[u8]) -> Result<Self> {
```

**Review by alamb — COMMENTED** [review]
Looks like it should work to me 👍  might be worth a test to document the expected behavior (perhaps as a follow on PR). I think we should merge this one asap to unblock IDPE / flux stuff

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
It might also be worth a ticket to track remove this old legacy format (filed under tech debt, perhaps)

**Comment by crepererum** [comment]
Removal ticket: https://github.com/influxdata/influxdb_iox/issues/4282

**Pull request merged**